### PR TITLE
Added some libc function

### PR DESCRIPTION
**Pull request by Supercip971**
- libc: added ctypes function
	- isspace isalnum isdigit isalpha to_lower to_upper
- libc: added strnlen
- libc: added strncmp
- libc: added memset
- libc: added memcpy

### CODE REVIEW

File: src/library/ctypes.c [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	return c;
+}
+
+int
+to_upper(int c)
+{
+	if (c >= 'a' && c <= 'z') {
+		c -= 'a';
+		c += 'A';
+	}
```
Comment (d0p1s4m4) [inline src/library/ctypes.c]: Why not using macro ?

File: src/library/string.c [hunk 2]
```diff
@@ -18,4 +28,41 @@ strcmp(const char *s1, const char *s2)
...
+	if (maxlen == 0) {
+		return 0;
+	}
+	return (int)s1[idx] - (int)s2[idx];
+}
+
+void *
+memset(void *s, int c, size_t n)
+{
+
```
Comment (d0p1s4m4) [inline src/library/string.c]: unneeded blank line

**Review by d0p1s4m4 — CHANGES_REQUESTED** [review]
also `{` are always on their one line. (check out change in .clang-format ) 

otherwise it seems perfect to me

File: src/library/string.c (hunk 2 above)
Comment (Supercip971) [reply src/library/string.c]: fixed <3

File: src/library/ctypes.c (hunk 1 above)
Comment (Supercip971) [reply src/library/ctypes.c]: in the standard they say it is a function not a macro

**Pull request merged**